src/ui/WcPanel.tsx

```diff
@@ -387,10 +430,139 @@ const WcPanel = ({ records, attributes, query, actions, adapter, creatingAdapter
 
   const saveAdapterCode = function() {
     chrome.storage.local.set({ [_adapterKey]: adapterCode }, function() {
-console.log("saved changes");
+    console.log("saved changes");
     });
   }
 
+  // This pattern matches every thing after and including any one of these symbols: = ( + - * /
+  const regex = /[=(\+\-\*\/][^=(\+\-\*\/]*$/; 
+    
+  // Teach Autosuggest how to calculate suggestions for any given input value.
+  const mathSymbols = {"Plus": "+", "Minus": "-", "Multiply": "*", "Divide": "/"};
+  const attributeNames = attributes.map(attribute => attribute.name);
+  const mathSuggestions = Object.keys(functions)
+    .filter(functionName => Object.keys(mathSymbols).indexOf(functionName) != -1)
+    .reduce((obj, functionName) => {
+      const symbol = mathSymbols[functionName];
+      obj[symbol] = functions[functionName];
+      return obj; 
+    }, {});
+  const allSuggestionNames = Object.keys(functions)
+    .sort()
+    .filter(functionName => ["Plus", "Minus", "Multiply", "Divide"].indexOf(functionName) == -1)
+    .concat(attributeNames);
+  const getSuggestions = value => {
+    const inputValue = value.trim()
+    const matchIndex = inputValue.search(regex);
+    
+    if (matchIndex == inputValue.length - 1) {
+      // If at the start of a new expression, include all possible suggestions
+      return allSuggestionNames.map(suggestion => inputValue + suggestion);
+    }
+    else if (matchIndex == -1) {
```

Comment:
Just a small general tip, I tend to prefer `===` for all equality checks in JS. For example this would return true if `matchIndex` was the string "-1", which seems unlikely in this case but can occasionally cause problems

Reply:
I think would be good to switch all of these to triple-equals unless you have a particular reason!